RasterPropMonitor/Core/StringProcessor.cs

```diff
@@ -19,124 +19,108 @@
  * along with RasterPropMonitor.  If not, see <http://www.gnu.org/licenses/>.
  ****************************************************************************/
 using System;
-using UnityEngine.Profiling;
+using System.Diagnostics;
 
 namespace JSI
 {
     public class StringProcessorFormatter
     {
-        // The formatString or plain text (if usesComp is false).
-        private readonly string formatString;
-        // An array of source variables
-        public readonly VariableOrNumber[] sourceVariables;
+        internal static readonly SIFormatProvider fp = new SIFormatProvider();
         // An array holding evaluants
         public readonly object[] sourceValues;
 
         public bool IsConstant => sourceValues == null;
 
         public string cachedResult;
 
-        // TODO: Add support for multi-line processed support.
-        public StringProcessorFormatter(string input, RasterPropMonitorComputer rpmComp)
+        // An array of source variables
+        internal readonly IVariable[] sourceVariables;
+        // The formatString or plain text (if usesComp is false).
+        private readonly string formatString;
+
+        public StringProcessorFormatter(string input, RasterPropMonitorComputer rpmComp, RasterPropMonitor rpm = null)
         {
-            if(string.IsNullOrEmpty(input))
+            if (string.IsNullOrEmpty(input))
             {
                 cachedResult = "";
+                return;
             }
-            else if (input.IndexOf(JUtil.VariableListSeparator[0], StringComparison.Ordinal) >= 0)
+
+            if (input.IndexOf(JUtil.VariableListSeparator[0], StringComparison.Ordinal) < 0)
             {
-                string[] tokens = input.Split(JUtil.VariableListSeparator, StringSplitOptions.RemoveEmptyEntries);
-                if (tokens.Length != 2)
+                cachedResult = input.TrimEnd();
+                return;
+            }
+
+            string[] tokens = input.Split(JUtil.VariableListSeparator, StringSplitOptions.RemoveEmptyEntries);
+            if (tokens.Length != 2)
+            {
+                throw new ArgumentException(string.Format("Invalid format string: {0}", input));
+            }
+
+            bool allVariablesConstant = true;
+
+            string[] sourceVarStrings = tokens[1].Split(JUtil.VariableSeparator, StringSplitOptions.RemoveEmptyEntries);
+            sourceVariables = new IVariable[sourceVarStrings.Length];
+            for (int i = 0; i < sourceVarStrings.Length; ++i)
+            {
+                var variableName = sourceVarStrings[i];
+                if (variableName.StartsWith("MONITOR_LOCAL_"))
```

Comment:
I think there's probably a much better way to implement monitor-local variables.